core/variables_dynamic.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Utility functions for handling variables dynamic.
+ *
+ * @author duzc2dtw@gmail.com (Du Tian Wei)
+ */
+'use strict';
+
+goog.provide('Blockly.VariablesDynamic');
+
+goog.require('Blockly.Variables');
+goog.require('Blockly.Blocks');
+goog.require('Blockly.constants');
+goog.require('Blockly.VariableModel');
+goog.require('Blockly.Workspace');
+goog.require('goog.string');
+
+
+Blockly.VariablesDynamic.onCreateVariableButtonClick_String = function(button) {
+  Blockly.Variables.createVariable(button.getTargetWorkspace(), null, 'String');
+};
+Blockly.VariablesDynamic.onCreateVariableButtonClick_Number = function(button) {
+  Blockly.Variables.createVariable(button.getTargetWorkspace(), null, 'Number');
+};
+Blockly.VariablesDynamic.onCreateVariableButtonClick_Color = function(button) {
+  Blockly.Variables.createVariable(button.getTargetWorkspace(), null, 'Color');
+};
+/**
+ * Construct the elements (blocks and button) required by the flyout for the
+ * variable category.
+ * @param {!Blockly.Workspace} workspace The workspace contianing variables.
+ * @return {!Array.<!Element>} Array of XML elements.
+ */
+Blockly.VariablesDynamic.flyoutCategory = function(workspace) {
+  var xmlList = [];
+  var button = goog.dom.createDom('button');
+  button.setAttribute('text', Blockly.Msg.NEW_VARIABLE);
```

Comment:
Please add new strings for these buttons that say "Create string variable", "Create number variable", and "Create color variable"